### PR TITLE
Prometheus bridge: Add tests to verify that start timestamps are working correctly

### LINKED ISSUE #4354: Prometheus bridge: Verify that prometheus start timestamps are working correctly
### Problem Statement

Now that support for creation timestamps exists in the promethues client, we should update to the latest version of the client, and verify that creation timestamps are working as expected.

https://github.com/prometheus/client_golang/releases/tag/v1.17.0

https://github.com/prometheus/client_golang/pull/1313

### Proposed Solution

This should be done with a new unit test.

### PR DESCRIPTION
**Pull request by hiroyaonoe**
fixes #4354 

I added tests to verify that the start timestamps of `counter` and `histogram` work properly.

If StartTime is set, StartTime minus processStartTime will be positive value.

I checked that these tests failed when I used the older version `v1.16.0` of `github.com/prometheus/client_golang`.

### CODE REVIEW

**Comment by hiroyaonoe** [comment]
> changelog / changelog (pull_request)

I think this is unnecessary because this PR is not a user-facing change.

> build_and_test / test-coverage (pull_request) 

This test failed due to https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4768 .

File: bridges/prometheus/producer_test.go [hunk 1]
```diff
@@ -251,3 +252,88 @@ func TestProduce(t *testing.T) {
...
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			reg := prometheus.NewRegistry()
+			tt.testFn(reg)
+			p := NewMetricProducer(WithGatherer(reg))
+			output, err := p.Produce(context.Background())
+			if tt.wantErr == nil {
+				assert.Nil(t, err)
+			}
```
Comment (dmathieu) [inline bridges/prometheus/producer_test.go:328]: `wantErr` is never used, and I'm not sure this test really ever needs to have an error on `Produce`. How about removing this option?

```suggestion
			assert.Nil(t, err)
```

Comment (pellared) [reply bridges/prometheus/producer_test.go:328]: (I changed the suggestion from `assert.Nil` to assert.NoError`)

**Review by dashpole — COMMENTED** [review]
Can you update the doc in bridges/prometheus/doc.go to remove the start time limitation?

**Comment by hiroyaonoe** [comment]
> wantErr is never used, and I'm not sure this test really ever needs to have an error on Produce. How about removing this option?

> Can you update the doc in bridges/prometheus/doc.go to remove the start time limitation?

I fixed both. Could you review it again?

File: bridges/prometheus/producer_test.go [hunk 2]
```diff
@@ -251,3 +252,85 @@ func TestProduce(t *testing.T) {
...
+		},
+	}
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			reg := prometheus.NewRegistry()
+			tt.testFn(reg)
+			p := NewMetricProducer(WithGatherer(reg))
+			output, err := p.Produce(context.Background())
+			assert.NoError(t, err)
+			for _, sms := range output {
```
Comment (dmathieu) [inline bridges/prometheus/producer_test.go:326]: This test would pass if output or their metrics aren't empty, yet there wouldn't be anything tested.

How about an assert.NotEmpty before each loop?

Comment (hiroyaonoe) [reply bridges/prometheus/producer_test.go:327]: I fixed it.
https://github.com/open-telemetry/opentelemetry-go-contrib/compare/849e455b517da4332ebe1a39506ddc858e422e2b..b125e660546eb80511082f56ef4f4cb400ebb56f

**Review by dmathieu — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Comment by hiroyaonoe** [comment]
Thank you for approving.
When is it merged?

**Pull request merged**